core/src/nats.rs

```diff
@@ -450,24 +450,29 @@ impl TypedNats {
         T: TypedMessage,
     {
         let bytes = Bytes::from(serde_json::to_vec(value)?);
-        let result1 = self
-            .nc
-            .request(value.subject(), bytes.clone());
-        
-        let result = if let Some(tmp_alt_subject) = value.tmp_alt_subject() {
-            let result2 = self.nc
-                .request(tmp_alt_subject, bytes);
-
-            select! {
-                result = result1 => result.to_anyhow()?,
-                result = result2 => result.to_anyhow()?,
-            }
-        } else {
-            result1.await.to_anyhow()?
+
+        let mut futs = FuturesUnordered::new();
+
+        futs.push(self.nc.request(value.subject(), bytes.clone()));
+        if let Some(tmp_alt_subject) = value.tmp_alt_subject() {
+            let result = self.nc.request(tmp_alt_subject, bytes);
+            futs.push(result);
         };
 
-        let value: T::Response = serde_json::from_slice(&result.payload)?;
-        Ok(value)
+        while let Some(result) = futs.next().await {
+            match result {
+                Ok(result) => {
+                    let value: T::Response = serde_json::from_slice(&result.payload)?;
+                    return Ok(value);
+                }
+                Err(err) => {
+                    tracing::warn!(error = ?err, "One future failed while waiting for response");
```

Comment:
Will we see this whenever there's no listener on the subject? In other words, we'll basically always see one of these warn messages since we only expect a listener to be on either the old _or_ the new subject?

Reply:
Usually, yes. The exception is that if we are running a wiretap, NATS doesn't know that the wiretap won't respond, so it will wait for some timeout before assuming that there are no responders.